src/instructions.js

```diff
@@ -4665,7 +4665,38 @@ t[0xF3] = cpu => {
     cpu.write_mmx64s(low, high);
 };
 
-t[0xF4] = cpu => { cpu.unimplemented_sse(); };
+t[0xF4] = cpu => {
+    cpu.task_switch_test_mmx();
+    cpu.read_modrm_byte();
+
+    if((cpu.prefixes & (PREFIX_MASK_REP | PREFIX_MASK_OPSIZE)) == PREFIX_66)
+    {
+        // pmuludq xmm1, xmm2/m128
+        let source = cpu.read_xmm_mem128s();
+        let destination = cpu.read_xmm128s();
+
+        let result_low = cpu.do_mul32(destination[0],source[0]);
+        let result_high = cpu.do_mul32(destination[2],source[2]);
```

Comment:
I don't think this works as the two `do_mul32`s alias their results (`do_mul32` doesn't do a copy). Could you add a regression test for this case?